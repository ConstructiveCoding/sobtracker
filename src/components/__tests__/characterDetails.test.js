import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import CharacterDetails from '../character/characterDetails';
import CharacterDetailsStyles from '../../theme/standard/components/characterDetails.styles';

describe('CharacterDetails', () => {
  it('should render the character details screen with the supplied props', () => {
    const styles = CharacterDetailsStyles.standard;

    const tree = renderer.create(
      <CharacterDetails
        name="Test Character"
        characterClass="Test Character Class"
        gender="Androgynous"
        level={0}
        gold={0}
        darkStone={0}
        initiative={0}
        corruption={0}
        git={0}
        style={styles}
      />
    );

    expect(tree).toMatchSnapshot();
  });
});
