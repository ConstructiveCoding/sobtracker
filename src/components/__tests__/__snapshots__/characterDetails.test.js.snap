// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetails should render the character details screen with the supplied props 1`] = `
<View
  style={
    Object {
      "paddingLeft": 16,
      "paddingRight": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 8,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Test Character
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 10,
          }
        }
      >
        Gender
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        Androgynous
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 10,
          }
        }
        testID="character-name"
      >
        Class
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        Test Character Class
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "textAlign": "left",
        }
      }
    >
      XP
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "textAlign": "right",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "textAlign": "left",
        }
      }
    >
      Level
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "textAlign": "right",
        }
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "textAlign": "left",
        }
      }
    >
      Gold
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "textAlign": "right",
        }
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "textAlign": "left",
        }
      }
    >
      Dark Stone
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "textAlign": "right",
        }
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "textAlign": "left",
        }
      }
    >
      Initiative
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "textAlign": "right",
        }
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "textAlign": "left",
        }
      }
    >
      Corruption
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "textAlign": "right",
        }
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "textAlign": "left",
        }
      }
    >
      Grit
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "textAlign": "right",
        }
      }
    />
  </View>
</View>
`;
